pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -96,37 +96,50 @@ func machinesEqual(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
 	return m1.Name == m2.Name && m1.Namespace == m2.Namespace
 }
 
+// isNodeJoin determines if a machine, in scope, should join of the cluster.
 func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachines []*clusterv1.Machine) (bool, error) {
 	switch set := scope.Machine.ObjectMeta.Labels["set"]; set {
 	case "node":
+		// Worker machines, not part of the controlplane, will always join the cluster.
 		return true, nil
 	case "controlplane":
+		// Controlplane machines will join the cluster if the cluster has an existing control plane.
+		controlplaneExists := false
+		var err error
 		for _, cm := range controlPlaneMachines {
 			m, err := actuators.NewMachineScope(actuators.MachineScopeParams{
-				Machine: cm,
-				Cluster: scope.Cluster,
-				Client:  a.clusterClient,
-				Logger:  a.log,
+				Machine:    cm,
+				Cluster:    scope.Cluster,
+				Client:     a.clusterClient,
+				Logger:     a.log,
+				AWSClients: scope.AWSClients,
 			})
 
 			if err != nil {
-				return false, errors.Wrapf(err, "failed to create machine scope for machine %q", cm.Name)
+				return false, errors.Wrapf(err, "failed to create machine scope for machine %q in namespace %q", cm.Name, cm.Namespace)
```

Comment:
Is this needed with the new logging? @chuckha can you confirm?

If it is still needed, I would suggest using `return false, errors.Wrapf(err, "failed to create machine scope for machine %s/%s", cm.Namespace, cm.Name)` instead.

Reply:
depends on if we log the error or not. I'd expect us to log the error which means this could just be a .Wrap call like `errors.Wrap(err, "failed to create a machine scope")`

Reply:
This error should be logged. I'll ensure it is and it is inline w/ @detiber's suggestion

Reply:
This error is wrapped in the machine actuator `Create` method so it should get logged.